packages/batch-submitter/src/transaction-chain-contract.ts

```diff
@@ -19,6 +20,29 @@ export { encodeAppendSequencerBatch, BatchContext, AppendSequencerBatchParams }
  * where the `appendSequencerBatch(...)` function uses a specialized encoding for improved efficiency.
  */
 export class CanonicalTransactionChainContract extends Contract {
+  public customPopulateTransaction = {
```

Comment:
We may want to name this differently to infer its purpose; e.g., `populateReplacementTransaction` or something?

Reply:
Note that this function isn't necessarily coupled to sending a replacement transaction, for that it would have have strict requirements around what the nonce is

Reply:
Ah yeah this populates both replacement transactions and the initial transaction that is sent.

Reply:
Ah yeah this populates both replacement transactions and the initial transaction that is sent.